src/FSharpPlus/Data/NonEmptyList.fs

```diff
@@ -156,16 +195,103 @@ module NonEmptyList =
 
 #endif
 
+    /// <summary>Returns a new list that contains all pairings of elements from two lists.</summary>
+    /// <param name="list1">The first input list.</param>
+    /// <param name="list2">The second input list.</param>
+    /// <returns>The resulting list of pairs.</returns>
+
+    let inline allPairs (list1: NonEmptyList<'T>) (list2: NonEmptyList<'U>) = Seq.allPairs list1 list2 |> ofSeq
+
+    /// <summary>Concatenates two lists.</summary>
+    /// <param name="list1">The first input list.</param>
+    /// <param name="list2">The second input list.</param>
+    /// <returns>The resulting list.</returns>
+    let inline append (list1: NonEmptyList<'T>) (list2: NonEmptyList<'T>) = 
+        { Head = list1.Head; Tail = list1.Tail @ list2.Head :: list2.Tail }
+
     /// <summary>Returns the average of the elements in the list.</summary>
     /// <param name="list">The input list.</param>
     /// <returns>The resulting average.</returns>
-    let inline average (list: NonEmptyList<'T>) = List.average (list.Head :: list.Tail)
+    let inline average (list: NonEmptyList<'T>) = Seq.average list
 
     /// <summary>Returns the average of the elements generated by applying the function to each element of the list.</summary>
     /// <param name="projection">The function to transform the list elements into the type to be averaged.</param>
     /// <param name="list">The input list.</param>
     /// <returns>The resulting average.</returns>
-    let inline averageBy (projection: 'T -> ^U) list = List.averageBy projection (list.Head :: list.Tail)
+    let inline averageBy (projection: 'T -> ^U)  (list: NonEmptyList<'T>) = List.averageBy projection (list.Head :: list.Tail)
```

Comment:
`NonEmptyList.average` uses `Seq.average` while `NonEmptyList.averageBy` uses `List.averageBy`, which seems inconsistent to me (I'm not very sure which is more performant, btw)

Reply:
`averageBy` was already implemented, so I mostly left it alone. I'll update it to match. 

I'm not sure which is more performant either. However, they should both by O(n).